refine.bat

```diff
@@ -188,6 +188,11 @@ if "%GDATA_CLIENT_SECRET%" == "" goto skipGDataCredentials
 set OPTS=%OPTS% -Dext.gdata.clientid=%GDATA_CLIENT_ID% -Dext.gdata.clientsecret=%GDATA_CLIENT_SECRET%
 :skipGDataCredentials
 
+if "%WIKIDATA_CLIENT_ID%" == "" goto skipWikidataCredentials
+if "%WIKIDATA_CLIENT_SECRET%" == "" goto skipWikidataCredentials
```

Comment:
Emm, the if statement doesn't support logical operators (https://stackoverflow.com/a/2143203/10464202), but we can rewrite it like 
```
if not "%WIKIDATA_CLIENT_ID%" == "" if not "%WIKIDATA_CLIENT_SECRET%" == "" set OPTS=%OPTS% -Dext.wikidata.clientid=%WIKIDATA_CLIENT_ID% -Dext.wikidata.clientsecret=%WIKIDATA_CLIENT_SECRET%
```
But that line will be too long.

Reply:
Oops, sorry! I was thinking of shell scripts and totally missed the fact that it's the Windows .bat file.

It's been so long since I've written Windows batch files that I'm not an expert at all, but if something like this would do the trick:

```
if not "%WIKIDATA_CLIENT_ID%" == "" (
  if not "%WIKIDATA_CLIENT_SECRET%" == "" (
    set OPTS=%OPTS% -Dext.wikidata.clientid=%WIKIDATA_CLIENT_ID% -Dext.wikidata.clientsecret=%WIKIDATA_CLIENT_SECRET%
  )
)
```

I think it would be slightly cleaner control flow, but don't spend more than 5 minutes on it.

Reply:
Terrific! I didn't notice that we can use parentheses here...